src/firmware/main/math/polynomial_1d.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+// Need this for size_t
+#include <stddef.h>
+
+/**
+ * A polynomial with a list of coefficients, with the highest order coefficients at
+ * the start of the list, ie:
+ * coefficients[0]*x^(order-1) + coefficients[1]*x^(order-2) + ... + coefficients[order-1]
```

Comment:
nit: shouldn't this start with `coefficients[0]*x^(order)` instead of `(order-1)`?

Reply:
Whoops, yep. Changed.